packages/editor/src/functions/takeScreenshot.ts

```diff
@@ -200,29 +200,29 @@ export async function takeScreenshot(
 
   // Rendering the scene to the new canvas with given size
   await new Promise<void>((resolve, reject) => {
+    // set up effect composer
+    EngineRenderer.instance.effectComposer.setMainCamera(scenePreviewCamera as Camera)
+    EngineRenderer.instance.effectComposer.setSize(width, height, false)
+    EngineRenderer.instance.renderer.setPixelRatio(1)
+
     const interval = setInterval(() => {
       const viewport = EngineRenderer.instance.renderContext.getParameter(
         EngineRenderer.instance.renderContext.VIEWPORT
       )
-      // todo - scrolling in and out sometimes causes weird pixel ratios that can cause this to fail
-      if (viewport[2] === Math.round(width * pixelRatio) && viewport[3] === Math.round(height * pixelRatio)) {
+      if (viewport[2] === Math.round(width) && viewport[3] === Math.round(height)) {
```

Comment:
Pixel ratio must be preserved here

Reply:
We are changing the pixel ratio to 1.
Also tried this a few times, `viewport` comes out to be like `0 0 512 320` and `height` and `width` are also `320` and `512` respectively but `pixelRatio` is `2`. So, the condition is not fulfilled. Also,`pixelRatio` is defined outside the promise so it does not change even though the promise inside changes the pixelRatio to 1.